strawberryfields/gbs/vibronic.py

```diff
@@ -0,0 +1,228 @@
+Designing a GBS device with such molecular parameters allows obtaining the Franck-Condon profile
+of the molecule which can be used to construct the molecular vibronic spectra.
+
+The Doktorov operator can be written in terms of displacement (:math:`D_\alpha`), squeezing
+(:math:`\Sigma`) and interferometer (:math:`C_L, C_R^t`) operators as
+
+.. math::
+    U_{Dok} = C_L \Sigma C_R^t D_\alpha.
+
+The squeezing and rotation operators are obtained by singular value decomposition of a matrix
+:math:`J` as
+
+.. math::
+    J = C_L \Sigma C_R^t.
+
+In order to obtain :math:`J` we need to construct diagonal matrices :math:`\Omega` and
+:math:`\Omega'` from square roots of the ground state and excited state normal mode frequencies,
+:math:`\omega` and :math:`\omega'`, respectively. Then :math:`J` is obtained as
+
+.. math::
+    J = \Omega' U \Omega^{-1},
+
+where
+
+.. math::
+    \Omega = diag (\sqrt{\omega_1},...,\sqrt{\omega_k})
+
+    \Omega' = diag (\sqrt{\omega_1'},...,\sqrt{\omega_k'})
+
+and :math:`U` is a molecular coordinate transformation known as the Duschinsky matrix which is
+obtained from the normal mode coordinates of the ground and excited states, :math:`q` and
+:math:`q'` respectively, as
+
+.. math::
+    q' = Uq + d.
+
+The displacement parameter :math:`d` is related to the structural changes of a molecule upon
+vibronic excitation and is related to the displacement parameter :math:`\alpha` as
+
+.. math::
+    \alpha = d / \sqrt{2}.
+
+In general, all the GBS parameters can be derived from the previously computed equilibrium
+geometries, vibrational frequencies and normal coordinates of the molecule in its (electronic)
+ground and excited states.
+
+
+Summary
+-------
+
+.. autosummary::
+    gbs_params
+    energy
+    spectrum
+
+Code details
+^^^^^^^^^^^^
+"""
+from typing import Tuple
+
+import numpy as np
+
+
+def gbs_params(
+    w: np.ndarray, wp: np.ndarray, Ud: np.ndarray, d: np.ndarray
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    r"""Converts molecular information input to Gaussian gate parameters.
+
+    We first construct diagonal matrices :math:`\Omega` and :math:`\Omega'` from square roots of
+    the ground state and excited state normal mode frequencies, :math:`\omega` and :math:`\omega'`,
+    respectively. Then we construct the matrix :math:`J` as
+
+    .. math::
+        J = \Omega' U \Omega^{-1},
+
+    where
+
+    .. math::
+        \Omega = diag (\sqrt{\omega_1},...,\sqrt{\omega_k}),
+
+        \Omega' = diag (\sqrt{\omega_1'},...,\sqrt{\omega_k'})
+
+    and :math:`U` is a molecular coordinate transformation known as the Duschinsky matrix.
+    The squeezing and rotation operators are obtained by singular value decomposition of the matrix
+    :math:`J` as
+
+    .. math::
+        J = C_L \Sigma C_R^t.
+
+    The displacement parameter :math:`\alpha` is obtained from the Duschinsky displacement
+    parameter :math:`d` as:
+
+    .. math::
+        \alpha = d / \sqrt{2}.
```

Comment:
Is this all a repetition of the module level docstring? We can probably just have a much shorter function docstring and leave all the explanation to the module level docstring.

Reply:
I will do that.